src/plugins/intel_cpu/src/nodes/executors/mlas/mlas_transpose.cpp

```diff
@@ -0,0 +1,309 @@
+            break;
+        }
+        case (sizeof(uint32_t)): {
+            SimpleTransposeSingleAxisInwards(reinterpret_cast<const uint32_t*>(input_data),
+                                                                             reinterpret_cast<uint32_t*>(output_data), num_loops, num_readers, reads_per_loop,
+                                                                             reads_per_reader_per_loop);
+            break;
+        }
+        case (sizeof(uint64_t)): {
+            SimpleTransposeSingleAxisInwards(reinterpret_cast<const uint64_t*>(input_data),
+                                                                             reinterpret_cast<uint64_t*>(output_data), num_loops, num_readers, reads_per_loop,
+                                                                             reads_per_reader_per_loop);
+            break;
+        }
+        default: {
+            // we need to use memcpy for each block
+            for (int64_t l = 0; l < num_loops; ++l) {
+                const uint8_t* input_for_first_reader = input_data;
+                for (auto rrpl = 0; rrpl < reads_per_reader_per_loop; ++rrpl) {
+                    const uint8_t* input_for_current_reader = input_for_first_reader;
+                    for (uint64_t r = 0; r < num_readers; ++r) {
+                        memcpy(output_data, input_for_current_reader, bytes_per_read);
+                        output_data += bytes_per_read;
+                        // skip to input position for next reader
+                        input_for_current_reader += (reads_per_reader_per_loop * bytes_per_read);
+                    }
+                    input_for_first_reader += bytes_per_read;
+                }
+                input_data += reads_per_loop * bytes_per_read;
+            }
+        }
+    }
+}
+
+void MlasTransposeExecutor::exec(const std::vector<MemoryCPtr>& src, const std::vector<MemoryPtr>& dst, const int MB) {
+    if (from > to) {
+            TransposeSingleAxisOutwards(src[0], dst[0], from, to);
+    } else {
+            TransposeSingleAxisInwards(src[0], dst[0], from, to);
+    }
+}
+
+bool MlasTransposeExecutor::init(const TransposeParams &transposeParams,
+                                 const std::vector<MemoryDescPtr> &srcDescs,
+                                 const std::vector<MemoryDescPtr> &dstDescs,
+                                 const dnnl::primitive_attr &attr) {
+    if (!IsTransposeMovingSingleAxis(transposeParams.permuteParams.order, from, to)) {
+        DEBUG_LOG("MLAS Transpose executor supports moving single axis only");
+        return false;
+    }
+    return true;
+}
+
+bool MlasTransposeExecutorBuilder::isSupported(const TransposeParams& transposeParams,
+                                               const std::vector<MemoryDescPtr>& srcDescs,
+                                               const std::vector<MemoryDescPtr>& dstDescs) const {
+    if (!srcDescs[0]->hasLayoutType(LayoutType::ncsp)) {
```

Comment:
1. dst layout should be the same is input one.
2. supported precision sizes (getPrecision.size()) are 1,2,4,8. We have to check this.